webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -469,9 +474,13 @@ trait CachedFindRuleNodeStatusReports
                                          .toIO
                        scoreUpdates <- ZIO.foreach(updated.toList) {
                                          case (nodeId, compliance) =>
-                                           val cp    = compliance.compliance.computePercent()
-                                           val event = ComplianceScoreEvent(nodeId, cp)
-                                           scoreServiceManager.handleEvent(event)
+                                           for {
+                                             userRules              <- rulesRepo.getIds(includeSytem = false)
+                                             complianceWithoutSystem = NodeStatusReport.filterByRules(compliance, userRules)
+                                             cp                      = complianceWithoutSystem.compliance.computePercent()
+                                             event                   = ComplianceScoreEvent(nodeId, cp)
+                                             _                      <- scoreServiceManager.handleEvent(event)
+                                           } yield ()
```

Comment:
since it's exactly the same code at two places and it's becoming non trivial, can you factor-it out somewhere ?

Reply:
Done!